lib/clients/transports/call-transport.js

```diff
@@ -85,24 +85,37 @@ var handleHttpErr = function handleHttpErr(retryOp, apiCb, statusCode) {
  * @param apiCb
  * @returns {boolean}
  */
-var handleHttpResponse = function handleHttpResponse(body, client, apiCb) {
+var handleHttpResponse = function handleHttpResponse(body, headers, client, apiCb) {
   var jsonResponse;
-  var jsonParseErr;
+  var jsonError;
 
   try {
     jsonResponse = JSON.parse(body);
   } catch (parseErr) {
     // TODO(leah): Emit an event here?
-    jsonParseErr = new Error('unable to parse Slack API Response');
+    jsonError = new Error('unable to parse Slack API Response');
     return false;
   }
 
-  if (!jsonParseErr && jsonResponse.warning) {
+  if (!jsonError && jsonResponse.warning) {
     client.logger('warn', jsonResponse.warning);
   }
 
+  if (!jsonResponse.ok) {
+    jsonError = new Error(jsonResponse.error);
+    client.logger('error', jsonResponse.error);
+  }
+
+  jsonResponse.scopes = (headers['X-OAuth-Scopes'] === undefined) ?
+    [] :
+    headers['X-OAuth-Scopes'].trim().split(/\s*,\s*/);
+  jsonResponse.acceptedScopes = (headers['X-Accepted-OAuth-Scopes'] === undefined) ?
+    [] :
+    headers['X-Accepted-OAuth-Scopes'].trim().split(/\s*,\s*/);
```

Comment:
Headers should be accessed in a case-insensitive way. If this is a node http.IncomingMessage then the headers will have been lowercased:

> Key-value pairs of header names and values. Header names are lower-cased. 
> -- https://nodejs.org/api/http.html#http_message_headers